thriftbp/client_middlewares.go

```diff
@@ -398,3 +400,28 @@ func PrometheusClientMiddleware(remoteServerSlug string) thrift.ClientMiddleware
 		}
 	}
 }
+
+func getClientError(result thrift.TStruct, err error) error {
+	if err != nil {
+		return err
+	}
+	v := reflect.ValueOf(result)
+	v = reflect.Indirect(v)
+	if v.Kind() != reflect.Struct {
+		return nil
+	}
+	resultField := v.FieldByName("Success").Interface()
+	for i := 0; i < v.NumField(); i++ {
+		field := v.Field(i).Interface()
+		if field == resultField {
```

Comment:
It looks like this code is trying to skip any struct fields named "Success".  Typically I would do that by `typ := v.Type(); ... if typ.Field(i).Name == "Success" { continue }`.

Right now this skips any field whose value matches the Success field, so if the success field is a boolean true, it'll skip any fields that are true.  This might function as desired but it's pretty weird, and may have unexpected panics on comparing types that can't be compared with `==`